doc/hooks/meson.section.md

```diff
@@ -1,17 +1,25 @@
 
 ### Meson {#meson}
 
-Overrides the configure phase to run meson to generate Ninja files. To run these files, you should accompany Meson with ninja. By default, `enableParallelBuilding` is enabled as Meson supports parallel building almost everywhere.
+Overrides the configure, check, and install phases to run `meson setup`, `meson test`, and `meson install`. For a meson-compatible build phase, you should accompany Meson with ninja. By default, `enableParallelBuilding` is enabled as Meson supports parallel building almost everywhere.
```

Comment:
It is not really obvious to me what Meson-compatible build phase is supposed to mean. Maybe something like the following:

```suggestion
Overrides the configure, check, and install phases to run `meson setup`, `meson test`, and `meson install`.

Meson is a meta-build system so you will need a secondary build system to run the generated build files in build phase. In Nixpkgs context, you will want to accompany Meson with ninja, which provides a [setup hook](#ninja) registering a ninja-based build phase.

By default, `enableParallelBuilding` is enabled as Meson supports parallel building almost everywhere.
```

Reply:
Incorporated.